drivers/rtc/Kconfig.numaker

```diff
@@ -0,0 +1,14 @@
+# NUMAKER RTC Driver configuration options
+
+# Copyright (c) 2024 Nuvoton Technology Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+config RTC_NUMAKER
+	bool "Nuvoton NuMaker MCU RTC driver"
+	default y
+	select HAS_NUMAKER_RTC
+	depends on DT_HAS_NUVOTON_NUMAKER_RTC_ENABLED
```

Comment:
If you take my comment https://github.com/zephyrproject-rtos/zephyr/pull/71929/files#r1579753122 into account, then you can have the `select` replaced with `depends on` and the `depends on DT_HAS_NUVOTON_NUMAKER_RTC_ENABLED` can be removed.

Reply:
Thanks of your opinion, DT_HAS_xx is asking DTS, and my thinking is like as:
[Kconfig.nrfx](https://github.com/zephyrproject-rtos/zephyr/blob/78150c8e1b2c07537ba35a05d9428983f1968f99/drivers/spi/Kconfig.nrfx#L15):
```
config SPI_NRFX_SPI
	def_bool y
	depends on DT_HAS_NORDIC_NRF_SPI_ENABLED
	select NRFX_SPI0 if HAS_HW_NRF_SPI0
```